tests/ImageSharp.Tests/Formats/Pbm/PbmDecoderTests.cs

```diff
@@ -0,0 +1,91 @@
+
+// ReSharper disable InconsistentNaming
+namespace SixLabors.ImageSharp.Tests.Formats.Pbm
+{
+    [Trait("Format", "Pbm")]
+    public class PbmDecoderTests
+    {
+        [Theory]
+        [InlineData(BlackAndWhitePlain, PbmColorType.BlackAndWhite)]
+        [InlineData(BlackAndWhiteBinary, PbmColorType.BlackAndWhite)]
+        [InlineData(GrayscalePlain, PbmColorType.Grayscale)]
+        [InlineData(GrayscaleBinary, PbmColorType.Grayscale)]
+        [InlineData(GrayscaleBinaryWide, PbmColorType.Grayscale)]
+        [InlineData(RgbPlain, PbmColorType.Rgb)]
+        [InlineData(RgbBinary, PbmColorType.Rgb)]
+        public void ImageLoadCanDecode(string imagePath, PbmColorType expectedColorType)
+        {
+            // Arrange
+            var testFile = TestFile.Create(imagePath);
+            using var stream = new MemoryStream(testFile.Bytes, false);
+
+            // Act
+            using var image = Image.Load(stream);
+
+            // Assert
+            Assert.NotNull(image);
+            PbmMetadata bitmapMetadata = image.Metadata.GetPbmMetadata();
+            Assert.NotNull(bitmapMetadata);
+            Assert.Equal(expectedColorType, bitmapMetadata.ColorType);
+        }
+
+        [Theory]
+        [InlineData(BlackAndWhitePlain)]
+        [InlineData(BlackAndWhiteBinary)]
+        [InlineData(GrayscalePlain)]
+        [InlineData(GrayscaleBinary)]
+        [InlineData(GrayscaleBinaryWide)]
+        public void ImageLoadL8CanDecode(string imagePath)
+        {
+            // Arrange
+            var testFile = TestFile.Create(imagePath);
+            using var stream = new MemoryStream(testFile.Bytes, false);
+
+            // Act
+            using var image = Image.Load<L8>(stream);
+
+            // Assert
+            Assert.NotNull(image);
+        }
+
+        [Theory]
+        [InlineData(RgbPlain)]
+        [InlineData(RgbBinary)]
+        public void ImageLoadRgb24CanDecode(string imagePath)
+        {
+            // Arrange
+            var testFile = TestFile.Create(imagePath);
+            using var stream = new MemoryStream(testFile.Bytes, false);
+
+            // Act
+            using var image = Image.Load<Rgb24>(stream);
+
+            // Assert
+            Assert.NotNull(image);
+        }
+
+        [Theory]
+        [WithFile(BlackAndWhiteBinary, PixelTypes.L8, true)]
+        [WithFile(GrayscalePlainNormalized, PixelTypes.L8, true)]
+        [WithFile(GrayscaleBinary, PixelTypes.L8, true)]
+        [WithFile(GrayscaleBinaryWide, PixelTypes.L16, true)]
+        [WithFile(RgbPlainNormalized, PixelTypes.Rgb24, false)]
+        [WithFile(RgbBinary, PixelTypes.Rgb24, false)]
+        public void DecodeReferenceImage<TPixel>(TestImageProvider<TPixel> provider, bool isGrayscale)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            using Image<TPixel> image = provider.GetImage();
+
+            image.CompareToReferenceOutput(provider, grayscale: isGrayscale);
```

Comment:
Helps reasoning about the output when testing locally with `Debug` builds:

```suggestion
            image.DebugSave(provider);
            image.CompareToReferenceOutput(provider, grayscale: isGrayscale);
```

Reply:
Sure, I interpreted this as left over debug code initially. Hence, I removed it before sending the pull request. Put them back now.

Reply:
`DebugSave()` is no-op when running Release configuration, so it doesn't hurt to keep them in all tests, but helps to see the output immediately in case a regression happens.